src/collectors/uniswapx_order_collector.rs

```diff
@@ -6,10 +6,53 @@ use reqwest::Client;
 use serde::Deserialize;
 use tokio::time::Duration;
 use tokio_stream::wrappers::IntervalStream;
+use std::fmt;
 
 static UNISWAPX_API_URL: &str = "https://api.uniswap.org/v2";
-static POLL_INTERVAL_SECS: u64 = 5;
-pub const CHAIN_ID: u64 = 1;
+static POLL_INTERVAL_SECS: u64 = 1;
+
+
+#[derive(Debug)]
+pub enum OrderTypeError {
+    InvalidOrderType,
+}
+
+impl fmt::Display for OrderTypeError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Invalid order type")
+    }
+}
+
+impl std::error::Error for OrderTypeError {}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum OrderType {
+    Dutch,
+    Priority,
+}
+
+impl OrderType {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            OrderType::Dutch => "Dutch_V2",
+            OrderType::Priority => "Priority",
+        }
+    }
+
+    pub fn from_str(s: &str) -> Result<OrderType, OrderTypeError> {
+        match s {
+            "Dutch_V2" => Ok(OrderType::Dutch),
+            "Priority" => Ok(OrderType::Priority),
+            _ => Err(OrderTypeError::InvalidOrderType),
+        }
+    }
```

Comment:
nit: use ToString and FromStr std traits
https://doc.rust-lang.org/std/str/trait.FromStr.html
https://doc.rust-lang.org/std/string/trait.ToString.html